Jint/Runtime/Interop/DefaultTypeConverter.cs

```diff
@@ -222,8 +222,23 @@ public virtual bool TryConvert(object value, Type type, IFormatProvider formatPr
 
             if (canConvert)
             {
-                converted = Convert(value, type, formatProvider);
-                return true;
+                //handle the common string to int conversion that otherwise removes the possibility to have overloaded indexers with both int and string 
+                var isStringValue = value is string;
```

Comment:
Couldn't we just use Convert.ChangeType() when we detect types are different? This would then also support other integral types